src/app/shared/services/api/search.repo.ts

```diff
@@ -68,19 +68,27 @@ export class SearchRepo extends BaseRepo {
     archiveId: string,
     limit?: number,
   ) {
-    const data = {
-      query,
-      tags,
-      archiveId,
-      publicOnly: true,
-    };
+    const endpoint = '/search/folderAndRecord';
+
+    const tagsQuery = tags
+      .map((tag, index) => `tags[${index}][tagId]=${tag.tagId}`)
+      .join('&');
+
+    const queryString =
+      `query=${query}&archiveId=${archiveId}&publicOnly=true` +
+      (limit ? `&numberOfResults=${limit}` : '') +
+      (tagsQuery ? `&${tagsQuery}` : '');
 
     return getFirst(
-      this.httpV2.get<SearchResponse>('/search/folderAndRecord', data, null, {
+      this.httpV2.get<SearchResponse>(`${endpoint}?${queryString}`, {}, null, {
         authToken: false,
       }),
     );
```

Comment:
This is a bit on me for using this code as an example of how we can use `const` in most places in JavaScript, but I didn't actually want my example to be directly used here. As mentioned in [this comment,](https://github.com/PermanentOrg/web-app/pull/490#discussion_r1848600610) we don't actually have to manually build up the query string as the HttpV2Service can take in a data object like a POST request, and generate and append a query string automatically.

My `const` suggestion was more a standard JavaScript tip, though it still had applicability in this function. Instead of building up a query string with `const`s, we should be building up a `const` data object we pass into the `httpV2.get` method. And since the parameters for this method are somewhat complex, we probably have to compose this data object out of multiple `const`s, or at least inlined expressions.

```ts
// I have not tested this code!
// This is just a sketch of what the code might look like.
// It definitely needs adjustments for typescript!

const tagQueries = tags.reduce((obj, tag, index) => {
    obj[`tags[${index}][tagId]`] = tag.tagId;
    return obj;
}, {});

const data = {
query,
archiveId,
publicOnly: 'true'
...tagQueries
};

if (limit) {
data.numberOfResults = limit;
}

return getFirst(
this.httpV2.get<SearchResponse>('/search/folderAndRecord', data, null, {
        authToken: false,
      })
);
```